From 0fbc0ecf76861a29b33502bd499e600e6f06a42c Mon Sep 17 00:00:00 2001
From: Jeremy Cline <jeremycline@microsoft.com>
Date: Wed, 18 Mar 2026 11:04:21 -0400
Subject: [PATCH] Limit list of elliptic curves tested to those in Fedora

The test currently fails on SECP192R1 and several other curves which
don't appear to be available in Fedora. Limit the test to the output of
"openssl ecparam -list_curves"
---
 tests/test_ssl.py | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/tests/test_ssl.py b/tests/test_ssl.py
index 4123377..2ed3cb9 100644
--- a/tests/test_ssl.py
+++ b/tests/test_ssl.py
@@ -1892,11 +1892,29 @@ class TestContext:
             "SECT571K1",
             "SECT571R1",
         }
+        # Based on the output of openssl ecparam -list_curves
+        # compared to dir(ec.EllipticCurveOID).
+        unavailable_in_fedora = [
+            'SECP192R1',
+            'SECP256R1',
+            'SECT163K1',
+            'SECT163R2',
+            'SECT233K1',
+            'SECT233R1',
+            'SECT283K1',
+            'SECT283R1',
+            'SECT409K1',
+            'SECT409R1',
+            'SECT571K1',
+            'SECT571R1'
+        ]
         for name in dir(ec.EllipticCurveOID):
             if name.startswith("_"):
                 continue
             if conftest.is_awslc and name in awslc_unsupported_curves:
                 continue
+            if name in unavailable_in_fedora:
+                continue
             oid = getattr(ec.EllipticCurveOID, name)
             cryptography_curve = ec.get_curve_for_oid(oid)
             context.set_tmp_ecdh(cryptography_curve())
-- 
2.53.0

